File: Stocker/ProjetoLDS/Stocker/Backend/StockerAPI/StockerAPI/Models/Dto/PurchaseDto.cs
using System.ComponentModel.DataAnnotations;

namespace StockerAPI.Models.Dto
{
    /// <summary>
    /// Representa os dados de uma compra, incluindo o grupo ao qual a compra pertence, a data e o valor total da compra.
    /// </summary>
    public class PurchaseDto
    {
        /// <summary>
        /// Identificador único da compra.
        /// Este campo é gerado automaticamente pelo sistema.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Identificador do grupo ao qual a compra pertence.
        /// Este campo é obrigatório e deve corresponder ao identificador de um grupo no sistema.
        /// </summary>
        [Required]
        public int Group_Id { get; set; }

        /// <summary>
        /// Data da compra.
        /// Este campo é obrigatório e deve refletir o momento em que a compra foi realizada.
        /// </summary>
        [Required]
        public DateTime Date { get; set; }

        /// <summary>
        /// Valor total da compra.
        /// Este campo é obrigatório e deve ser um valor maior ou igual a zero.
        /// </summary>
        [Required]
        [Range(0.0f, float.MaxValue, ErrorMessage = "Preço deve ser maior ou igual a 0.")]
        public float Price { get; set; }
    }
}

File: Stocker/ProjetoLDS/Stocker/Frontend/StockerMobileUI/stocker_web_ui/lib/models/PurchaseDto.dart
import 'dart:ffi';

/// Classe `Purchasedto`
///
/// Representa uma compra, contendo informações como identificador, grupo,
/// data da compra e preço total.
class Purchasedto {
  /// Identificador único da compra.
  final int Id;

  /// Identificador do grupo associado à compra.
  final int Group_Id;

  /// Data da compra.
  final String Date;

  /// Preço total da compra.
  final double Price;

  /// Construtor da classe `Purchasedto`.
  ///
  /// - [Id]: Identificador único da compra.
  /// - [Group_Id]: Identificador do grupo associado à compra.
  /// - [Date]: Data da compra.
  /// - [Price]: Preço total da compra.
  const Purchasedto({
    required this.Id,
    required this.Group_Id,
    required this.Date,
    required this.Price,
  });

  /// Converte um JSON num objeto `Purchasedto`.
  factory Purchasedto.fromJson(Map<String, dynamic> json) => Purchasedto(
      Id: json["id"],
      Group_Id: json["group_Id"],
      Date: json["date"],
      Price: (json["price"] as num).toDouble());

  /// Converte um objeto `Purchasedto` num JSON.
  Map<String, dynamic> toJson() => {
        "id": Id,
        "group_Id": Group_Id,
        "date": Date,
        "price": Price,
      };
}
